blockchain/chain/src/store/chain_store.rs

```diff
@@ -248,6 +265,43 @@ where
     Ok(())
 }
 
+/// Gets 32 bytes of randomness for ChainRand paramaterized by the DomainSeparationTag, ChainEpoch, Entropy
+pub fn get_randomness<DB: BlockStore>(
+    db: &DB,
+    blocks: &TipsetKeys,
+    pers: DomainSeparationTag,
+    round: ChainEpoch,
+    entropy: &[u8],
+) -> Result<[u8; 32], Box<dyn std::error::Error>> {
+    let mut blks = blocks.clone();
+    loop {
+        let nts = tipset_from_keys(db, &blks)?;
+        let mtb = nts.min_ticket_block();
+        if nts.epoch() <= round || mtb.epoch() == 0 {
+            return draw_randomness(mtb.ticket().vrfproof.bytes(), pers, round, entropy);
+        }
+        blks = mtb.parents().clone();
+    }
+}
+
+/// Computes a pseudorandom 32 byte Vec
+fn draw_randomness(
+    rbase: &[u8],
+    pers: DomainSeparationTag,
+    round: ChainEpoch,
+    entropy: &[u8],
+) -> Result<[u8; 32], Box<dyn std::error::Error>> {
+    let mut state = Params::new().hash_length(32).to_state();
+    state.write_i64::<BigEndian>(pers as i64)?;
```

Comment:
can you make the DomainSeparationTag repr a i64 just to ensure this is consistent?